### DISCUSSION TITLE
Sideband Settings

### DISCUSSION THREAD

**Discussion by 8uu8** [markqvist/Reticulum#595] [Help & Questions] [answered]
Help you understand the connection for communication and messaging in the app Sideband. After installing on two android phones, how to start communicating via the Internet or Wi-Fi? Add a contact? Do I need to exchange keys? Nowhere did I find such detailed instructions

**Comment by cobraPA** [comment, marked as answer]
Did you find the getting started guide?  I forget if it shows:

1. Setup both devices on a common network (wifi is easiest, so they can directly see each other.)
2. When you launch sideband, look at the announces page on the other device.  Normally they will auto announce their ID.
3. If left for a while, they will auto announce again.  Or you can press the announce icon ((.)) to announce now.
4. Now both should have a visible peer in the announces screen, so just reply to that ID to send messages between them.

**Reply by cobraPA** [reply]
For configuration beyond direct peer-to-peer on the local subnet, the reticulum manual goes in to all the details, I think.  https://github.com/markqvist/Reticulum

**Reply by 8uu8** [reply]
Thank you for participating! The link you provided provides general instructions. Still, I would not refuse a more detailed explanation

**Reply by controlrepo** [reply]
Hey,

I'm surprised by the selected answer.

I have similar issues as the OP. I am a reasonably experienced sysadmin with some networking experience and I struggle to make three Android phones (two 1.5.0, one 1.1.13) to talk to each other or to anything else in that matter.

Comments to the points of @cobraPA response.
1. all the networks I have available have client isolation turned on; long term kex over ad-hoc wifi networks OR NFC would probably be the best (I can't imagine people at the protest trying to huddle by the local Starbucks and trying to send the messages to each other over the public wifi. Almost all the public wifi networks, majority of the local networks have client isolation on anyway. Either in the AP directly or ACLs)
2. none seen on any of the phones
3. well, it auto announces anyway, but nothing is ever seen
4. nope.

I tried turning on the public gateways in the Connectivity menu, "connect via TCP".
On one phone I tried the default `sideband.connect.reticulum.network:7822`, on another two I tried different entries from [this file](https://github.com/markqvist/reticulum_website/blob/main/constants.py) and the Network's [suggested Testnet entrypoints](https://reticulum.network/connect.html), turning it off and on again, even trying to restart the app.

Of course there's no indication what "Optional IFAC network name/passphrase means", so I left it blank as it was.

To no avail.
`Connectivity status` shows `0 reachable peers` on all.
`Full RNS Status` shows Status: Up and Peers: 0 reachable....

> For configuration beyond direct peer-to-peer on the local subnet

I would argue that should be a *PRIMARY* goal/setup of the Sideband, otherwise it's just a useless time sink and can be replaced with *ANY* direct communication device, like dMPR Tier 2 radios -- which will additionally offer, out of the box, comparable if not better range (depending on the antenna).

As I understand Reticulum is a *network* "for building local and wide-area networks"
If Sideband is advertised as an app working on top of Reticulum, "wide-area networks" should not be an optional afterthought.

> For configuration beyond direct peer-to-peer on the local subnet, the reticulum manual goes in to all the details, I think. https://github.com/markqvist/Reticulum

No, it doesn't, not at all.
I guess you didn't check if the link you provided contains the information you suggest it does.

First I checked the [FAQ](https://github.com/markqvist/Reticulum/wiki/Frequently-Asked-Questions), as I though the flagship (?) app basic setup could be referred there, but Sideband is only mentioned by name.

The main repository you linked lists [markqvist.github.io/Reticulum/manual/](https://markqvist.github.io/Reticulum/manual/) as the manual URL. 
It talks almost exclusively about - predictably - Reticulum.
[Getting started fast](https://markqvist.github.io/Reticulum/manual/gettingstartedfast.html) once again only *[MENTIONS](https://markqvist.github.io/Reticulum/manual/gettingstartedfast.html#sideband)* Sideband by name, offers 2 sentences of blurb and 3 generic screenshots.

One of the two sentences about Sideband links to the [Sideband website](https://unsigned.io/website/sideband/) 

It contains the extended, _press release_ I'd say, going in lengths on functionality, the same three screenshots, says where to download/install APK from and offers a link to the [Guides](https://unsigned.io/website/guides.html).

Guides offer no guidance on configuring Sideband other than with the dedicated local hardware.

And that's all really.

> For configuration beyond direct peer-to-peer on the local subnet, the reticulum manual goes in to all the details

No it doesn't, not in the direction asked by the @8uu8 :

>> Help you understand the connection for communication and messaging in the app Sideband. After installing on two android phones, how to start communicating via the Internet or Wi-Fi? Add a contact? Do I need to exchange keys? Nowhere did I find such detailed instructions

This is not covered anywhere and tbh I wouldn't accept your comment as an answer.

**Comment by LinuxinaBit** [comment]
I understand the confusion. Sideband's UI is definitely the worst of the three major LXMF clients.
That being said, I'll try to help as best as I can.

> How to start communicating via the Internet or Wi-Fi?

Local WiFi connections should be available automatically (as it's enabled by default) so long as your router doesn't block IPv6 connections between devices on said local network. Same for other LAN connections as well.

As for over the internet, and/or IPv4, Sideband doesn't allow much configuration for this from the GUI like Reticulum MeshChat does; in Sideband you can only easily configure one TCPclientinterface connection - but it should be serviceable for most things.

1. Open the sidebar.
2. Click `Connectivity`
3. Enable `Connect via TCP`
4. Change the host and port if you want to connect somewhere else (the default values will connect you to the testnet - you probably want that)
5. _**Click the `Shutdown` button in the sidebar to apply the changes. Simply swiping away the app will not work.**_

> Add a contact?

1. Make sure both devices have some path to each-other and have sent announces by clicking the announce button on the `Conversations` page: <img src="https://github.com/user-attachments/assets/5b27fe33-ba1b-4863-94e7-99c5f858ad4c" height="18px"> ([why?](https://linuxinabit.codeberg.page/blog/reticulum/#How_Announces_Work))
2. If you **do not** have the other device's address, you may be able to find it on the `Announce Stream` page. To start a chat with one, use the three-dot button <img src="https://github.com/user-attachments/assets/9573a6c3-f3f1-4755-80f0-5ef7046f7058" height="18px"> and tap `Converse`. You can change your device's name from the `Preferences` page (make sure to `Shutdown` if you do)
3. If you **do** have their address, click the 'add user' <img src="https://github.com/user-attachments/assets/1cc7d5c6-4b6a-4ce0-a78b-ce9aed00d7a3" height="18px"> button on the `Conversations` page and paste the address there (you can find a device's address in the `Preferences` page - double tap quickly to copy)

> Do I need to exchange keys?

No. The network should handle this for you, even if you do not have an announce from the other device.

> Nowhere did I find such detailed instructions

Yeah, Reticulum is in that awkward development state where nothing is really comprehensively documented but it's working well enough that people actually want to use it. It's difficult and I 100% get it.

Hopefully this helped at least a little. If you need more real-time support, you can always ask in the [Matrix room](https://matrix.to/#/#reticulum:matrix.org) as well.

**Comment by elzorrofxt** [comment]
tldr; check your Windows Firewall settings

I had problems with this recently.  When I installed and ran the Reticulum MeshChat software on Win11 I was prompted to allow the software to connect (usual Windows Firewall prompt).  I accepted.  Couldn't connect to a second device (Android running Sideband) that was only "local Wifi/Ethernet" with TCP disabled.  Checking the Windows Firewall on the first machine I could see that Reticulum was allowed "public" but not "private".  Gave it "private" as well and the two were able to connect just fine and I can see the second machine in the Network Visualizer.
